mlflow/pyfunc/__init__.py

```diff
@@ -2429,12 +2429,19 @@ def predict(model_input: List[str]) -> List[str]:
                 CHAT_MODEL_INPUT_SCHEMA,
                 CHAT_MODEL_OUTPUT_SCHEMA,
             )
-            input_example = CHAT_MODEL_INPUT_EXAMPLE
-
-            # perform output validation and throw if
-            # output is not coercable to ChatResponse
-            messages = [ChatMessage(**m) for m in input_example["messages"]]
-            params = ChatParams(**{k: v for k, v in input_example.items() if k != "messages"})
+            input_example = input_example or CHAT_MODEL_INPUT_EXAMPLE
+
+            if isinstance(input_example, list):
+                params = ChatParams()
+                messages = [
+                    each_message
+                    for each_message in input_example
+                    if isinstance(each_message, ChatMessage)
```

Comment:
Users will typically pass in a list of dictionaries for multi-message input examples, not `ChatMessage` objects. Can we support that case too?

When `input_example` is a list of `ChatMessage`, what does the `input_example.json` file end up looking like? Does `ChatMessage` get serialized to JSON correctly?

Reply:
> When input_example is a list of ChatMessage, what does the input_example.json file end up looking like? Does ChatMessage get serialized to JSON correctly?


Yes there is already a test for that added. https://github.com/mlflow/mlflow/pull/12115/files#diff-de48be7bacec8de988bc9f6f1287dc1ddec2f5b0a7aa3eae71813070485aefb0R280


> Users will typically pass in a list of dictionaries for multi-message input examples, not ChatMessage objects. Can we support that case too?
Is this this format?
```
 input_example = {
     messages: {
          ChatMessage(role="user", content="What is Retrieval-augmented Generation?", name="chat"),
          ChatMessage(role="user", content="What is Retrieval-augmented Generation?", name="chat")
    }
}
```

Reply:
`input_example = {
    "messages": [
        {
            "role": "...",
            "content": "..."
        },
        {
            "role": ...,
            "content": ...
        }
    ]